src/Button/index.js

```diff
@@ -224,11 +224,16 @@ export const StyledButton = styled(
     block,
     style,
     dataTest,
+    submit,
     ...props
   }) => {
     const Component = component === "button" && props.href ? "a" : component;
     return (
-      <Component data-test={dataTest} {...props}>
+      <Component
+        data-test={dataTest}
```

Comment:
![image](https://user-images.githubusercontent.com/7850592/46398178-b07b7180-c6f4-11e8-8ee7-e8942b33f9f8.png)

Can we do simple refactor ? this is nicer

Reply:
Needs to be
```jsx
type={!isButtonWithHref ? (submit && "submit") || "button" : undefined}
```
otherwise we are rendering type="false" to the DOM